pkg/apis/kubevirt/v1alpha1/types_worker.go

```diff
@@ -26,6 +27,12 @@ import (
 type WorkerConfig struct {
 	metav1.TypeMeta `json:",inline"`
 
+	// MemoryFeatures allows specifying the VirtualMachineInstance memory features like huge pages and guest memory settings.
+	// Each feature might require appropriate FeatureGate enabled.
+	// For hugepages take a look at:
+	// k8s - https://kubernetes.io/docs/tasks/manage-hugepages/scheduling-hugepages/
+	// okd - https://docs.okd.io/3.9/scaling_performance/managing_hugepages.html#huge-pages-prerequisites
+	MemoryFeatures *kubevirtv1.Memory `json:"memoryFeatures,omitempty"`
```

Comment:
You may also need to set memory request here too, though the [doc ](https://kubevirtlegacy.gitbook.io/user-guide/creation/devices#limitations) doesn't say what is the consequence of empty memory request.

Reply:
> You may also need to set memory request here too

thanks for your input @rootfs, we do set memory request but in a bit different place to align with gardener API. It's passed from a `machineType` object https://github.com/gardener/gardener-extension-provider-kubevirt/blob/master/pkg/controller/worker/machines.go#L149. 

> though the doc doesn't say what is the consequence of empty memory request.

KubeVirt validates if your hugepages size is bigger than requested memory, so in case it's empty KubeVirt would return an error.